File: app/src/main/java/com/absinthe/anywhere_/adapter/manager/WrapContentLinearLayoutManager.java
package com.absinthe.anywhere_.adapter.manager;

import android.content.Context;

import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import timber.log.Timber;

public class WrapContentLinearLayoutManager extends LinearLayoutManager {
    public WrapContentLinearLayoutManager(Context context) {
        super(context);
    }

    public WrapContentLinearLayoutManager(Context context, @RecyclerView.Orientation int orientation) {
        super(context, orientation, false);
    }

    @Override
    public void onLayoutChildren(RecyclerView.Recycler recycler, RecyclerView.State state) {
        try {
            super.onLayoutChildren(recycler, state);
        } catch (IndexOutOfBoundsException e) {
            Timber.e("encounter an IOOBE in RecyclerView");
        }
    }
}

File: app/src/main/java/com/absinthe/anywhere_/adapter/manager/WrapContentLinearLayoutManager.kt
package com.absinthe.anywhere_.adapter.manager

import android.content.Context
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.Recycler
import timber.log.Timber

class WrapContentLinearLayoutManager : LinearLayoutManager {

    constructor(context: Context?) : super(context)

    constructor(context: Context?, @RecyclerView.Orientation orientation: Int) : super(context, orientation, false)

    override fun onLayoutChildren(recycler: Recycler, state: RecyclerView.State) {
        try {
            super.onLayoutChildren(recycler, state)
        } catch (e: IndexOutOfBoundsException) {
            Timber.e("encounter an IOOBE in RecyclerView")
        }
    }
}